### ISSUE TITLE
ContractAgreement: store them separately from negotiation

### ISSUE THREAD

**Issue by ndr-brt** [eclipse-dataspaceconnector/DataSpaceConnector#972]
# Feature Request

Introduce a `ContractAgreementStore` and use it to persist contract agreements.

## Which Areas Would Be Affected?
`contract`, in memory, cosmos and sql implementations

## Why Is the Feature Desired?
Currently `ContractAgreement`s are stored inside the negotiation.
This make pretty difficult/impossibile to get all the agreement on a particular contract definition.
Having this would be pretty important to permit checking before the deletion of a definition that is already referenced by a contract agreement (see #940)

## Solution Proposal
-

## Type of Issue
improvement

## Checklist

- [x] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

**Comment by jimmarino** [comment]
Why not model this on the ContractNegotiationStore with a separate retrieval method? A ContractAgreement cannot exist without a ContractNegotiation.

**Comment by ndr-brt** [comment]
yes, by its id, not by definition id, that's a part of the agreement id.
probably a separate retrieval method on contract negotiation would work too, but in every case separating the storage could give some advantages, one in particular is to have a simpler implementation/less load on the negotiation store for the `getAll` endpoint on the `ContractAgreementApi`:

https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/extensions/api/data-management/contractagreement/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractagreement/ContractAgreementApi.java

**Comment by ndr-brt** [comment]
Superseded by #1008
